### PR TITLE
Ah/more tests

### PR DESCRIPTION
**Pull request by anthonyholley**
Still working on adding more, I think all the basics are covered here though

### CODE REVIEW

**Comment by Howon** [comment]
Do these parse correctly?

**Comment by anthonyholley** [comment]
I've tried parsing some, all successfully, but didn't go through all of them. I think it would be more productive for me to get more test files made right now, but happy to try to parse the ones I haven't checked yet instead if you think that's better?

**Comment by jibben** [comment]
please make sure all of your tests parse

File: .gitignore [hunk 1]
```diff
@@ -1,15 +0,0 @@
-*~
```
Comment (jibben) [inline .gitignore:1]: ...did you remove .gitignore?

File: test/oscar/compiler/setIntersect.oscar.out [hunk 2]
```diff
@@ -0,0 +1 @@
+[4,5]
```
Comment (jibben) [inline test/oscar/compiler/setIntersect.oscar.out]: sets ordering is indeterminate, so you can't check this way

Comment (anthonyholley) [reply test/oscar/compiler/setIntersect.oscar.out]: Ah thank you, on it rn

File: test/oscar/compiler/list.oscar [hunk 3]
```diff
@@ -0,0 +1,6 @@
+def main() => unit = {
+  println(list<int>[]);
+  println(list<int>[1,2,3,4,5]);
+  println(list<double>[6.0,7.0,8.0])
+  println(list<double> (10, 0.0));
```
Comment (jibben) [inline test/oscar/compiler/list.oscar]: please remove this from the test case

Comment (anthonyholley) [reply test/oscar/compiler/list.oscar]: Everything?

Comment (jibben) [reply test/oscar/compiler/list.oscar]: just this line

Comment (anthonyholley) [reply test/oscar/compiler/list.oscar]: there are 4 lines there

File: test/oscar/compiler/list.oscar [hunk 4]
```diff
@@ -0,0 +1,6 @@
+def main() => unit = {
+  println(list<int>[]);
```
Comment (jibben) [inline test/oscar/compiler/list.oscar]: not this

File: test/oscar/compiler/list.oscar [hunk 5]
```diff
@@ -0,0 +1,6 @@
...
+  println(list<int>[1,2,3,4,5]);
```
Comment (jibben) [inline test/oscar/compiler/list.oscar]: not this

File: test/oscar/compiler/list.oscar [hunk 6]
```diff
@@ -0,0 +1,6 @@
...
+  println(list<double>[6.0,7.0,8.0])
```
Comment (jibben) [inline test/oscar/compiler/list.oscar]: not this

File: test/oscar/compiler/list.oscar (hunk 3 above)
Comment (jibben) [reply test/oscar/compiler/list.oscar]: just this

File: test/oscar/compiler/elseIf2.oscar [hunk 7]
```diff
@@ -0,0 +1,11 @@
+def main() => unit = {
+  if (1 > 2) {
+    println("if4 broken");
+  } else if (1 > 3) {
```
Comment (jibben) [inline test/oscar/compiler/elseIf2.oscar]: no else if

File: test/oscar/compiler/doubleDivision.oscar [hunk 8]
```diff
@@ -0,0 +1,3 @@
+def main() => unit = {
+  println(1.1 / 2.);
```
Comment (jibben) [inline test/oscar/compiler/doubleDivision.oscar]: There's an edge case sometimes with doubles. You have to compare in a delta like this:

(1.1 / 2. > 0.549) && (1.1 / 2. < 0.551)

File: test/oscar/compiler/AND.oscar.out [hunk 9]
```diff
@@ -0,0 +1,4 @@
+true 
+false 
+false 
+false
```
Comment (vlad-scherbich) [inline test/oscar/compiler/AND.oscar.out]: This is going to be annoying, but ... the output files that are attached to tests which use `Println` (like this one) might need newlines added at the end.

This is simply b/c `Println` appends a newline to the end of its output, which means that even though the output file content is correct, it won't find the last newline and fail the test.

We should test this to make sure before doing work, but that's my feel on this.

File: test/oscar/compiler/boolMutable.oscar [hunk 10]
```diff
@@ -0,0 +1,19 @@
+message testNow()
+
+
+actor Tester() {
+  mut bool b = false;
+  
+  receive = {
+    | testNow() => {
+        Println(b);
+		b = true;
```
Comment (vlad-scherbich) [inline test/oscar/compiler/boolMutable.oscar]: please indent these two lines

File: test/oscar/compiler/charMutable.oscar [hunk 11]
```diff
@@ -0,0 +1,19 @@
+message testNow()
+
+
+actor Tester() {
+  mut char c = 'a';
+  
+  receive = {
+    | testNow() => {
+        Println(c);
+		c = 'b';
```
Comment (vlad-scherbich) [inline test/oscar/compiler/charMutable.oscar:10]: indent here

File: test/oscar/compiler/doubleMutable.oscar [hunk 12]
```diff
@@ -0,0 +1,19 @@
+message testNow()
+
+
+actor Tester() {
+  mut double x = 0.0;
+  
+  receive = {
+    | testNow() => {
+        Println(x > -0.000001);
+		x = x - 0.000001;
```
Comment (vlad-scherbich) [inline test/oscar/compiler/doubleMutable.oscar:10]: indent here

File: test/oscar/compiler/AND.oscar.out (hunk 9 above)
Comment (anthonyholley) [reply test/oscar/compiler/AND.oscar.out]: No worries, will get the extra lines added in

**Review by jibben — APPROVED** [review]
